bubblesub/ui/views.py

```diff
@@ -0,0 +1,110 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from PyQt5 import QtCore, QtWidgets
+
+from bubblesub.api import Api
+from bubblesub.api.gui import ViewState
+from bubblesub.ui.audio import Audio
+from bubblesub.ui.console import Console
+from bubblesub.ui.editor import Editor
+from bubblesub.ui.video import Video
+
+
+class View(QtCore.QObject):
+    def __init__(
+        self,
+        api: Api,
+        video: Video,
+        audio: Audio,
+        editor: Editor,
+        console: Console,
+        main_window: QtWidgets.QMainWindow,
+    ) -> None:
+        super().__init__()
+
+        self._api = api
+        self._video = video
+        self._audio = audio
+        self._editor = editor
+        self._console = console
+        self._main_window = main_window
+        self._state = ViewState.Full
+
+    @property
+    def view_state(self) -> ViewState:
+        return self._state
+
+    def audio_view(self) -> None:
+        self._state = ViewState.Audio
+        self._api.playback.is_paused = True
+
+        self._enable_widgets([True, False, False])
+        self._show_widgets([True, False, False])
+
+    def video_view(self) -> None:
+        self._state = ViewState.Video
+        self._api.playback.is_paused = True
+
+        self._enable_widgets([False, True, False])
+        self._show_widgets([False, True, False])
+
+    def subs_view(self) -> None:
+        self._state = ViewState.Subs
+        self._api.playback.is_paused = True
+
+        self._enable_widgets([False, False, False])
+        self._show_widgets([False, False, False])
+
+    def full_view(self) -> None:
+        self._state = ViewState.Full
+        self._api.playback.is_paused = True
+
+        self._enable_widgets([True, True, True])
+        self._show_widgets([True, True, True])
+
+    def restore_view(self) -> None:
+        view = self._api.cfg.opt["view"]["current"]
+        func = {
+            "Full": self.full_view,
+            "Audio": self.audio_view,
+            "Video": self.video_view,
+            "Subs": self.subs_view,
+        }.get(view)
+
+        if func is not None:
+            func()
+
+    def store_view(self) -> None:
+        view = {
+            ViewState.Full: "Full",
+            ViewState.Audio: "Audio",
+            ViewState.Video: "Video",
+            ViewState.Subs: "Subs",
+        }.get(self._state)
+
+        self._api.cfg.opt["view"]["current"] = view
+
+    def _enable_widgets(self, value: [bool, bool, bool]) -> None:
+        self._audio.setEnabled(value[0])
+        self._video.setEnabled(value[1])
+        self._console.setEnabled(value[2])
```

Comment:
This function should be merged with `_show_widgets`, as they're called exactly the same way